### PR TITLE
Add some tests for copy behavior

### PR DESCRIPTION
**Pull request by gsoltis**
In preparation for adding a filesystem abstraction.

### CODE REVIEW

File: cli/internal/cache/cache_fs_test.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+	var tt interface{} = t
+	if tu, ok := tt.(testingUtil); ok {
+		tu.Cleanup(func() { _ = os.RemoveAll(dir) })
+	}
+}
+
+func TestPut(t *testing.T) {
+	src := subdirForTest(t)
+
+	childDir := filepath.Join(src, "child")
```
Comment (jaredpalmer) [inline cli/internal/cache/cache_fs_test.go:46]: Add a comment explaining the setup here

File: cli/internal/cache/cache_fs_test.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+	}
+}
+
+func TestFetch(t *testing.T) {
+	cacheDir := subdirForTest(t)
+	src := filepath.Join(cacheDir, "the-hash", "some-package")
+	err := os.MkdirAll(src, os.ModeDir|0777)
+	assert.NilError(t, err, "mkdirAll")
+
+	childDir := filepath.Join(src, "child")
```
Comment (jaredpalmer) [inline cli/internal/cache/cache_fs_test.go:127]: Add a comment explaining the setup

**Review by jaredpalmer — CHANGES_REQUESTED** [review]

File: cli/internal/cache/cache_fs_test.go (hunk 1 above)
Comment (gsoltis) [reply cli/internal/cache/cache_fs_test.go:65]: Done

File: cli/internal/cache/cache_fs_test.go (hunk 2 above)
Comment (gsoltis) [reply cli/internal/cache/cache_fs_test.go:166]: Done

**Comment by gsoltis** [comment]
Also added some more assertions to round out the `Fetch` test.

**Review by jaredpalmer — APPROVED** [review]

**Pull request merged**